File: C++/프로그래머스/0/120880. 특이한 정렬/특이한 정렬.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> numlist, int n) {
    vector<int> answer(numlist);
    
    //sort 사용. lambda 함수
    sort(answer.begin(), answer.end(), [n](int a, int b) {
        // cmath의 절대값 함수 사용
        int na = abs(n - a), nb = abs(n - b);
        
        // n과 직선좌표계상 거리가 같으면
        if (na == nb) return a > b;
        // 아니면 거리 순 정렬
        return na < nb;
    });
    
    return answer;
}

File: Go/프로그래머스/0/120880. 특이한 정렬/특이한 정렬.go
import "sort"

func Distance(num, std int) int {
    n := std - num
    if n < 0 {
        return -n
    }
    return n
}

func solution(numlist []int, n int) []int {
    copied := make([]int, len(numlist))
    copy(copied, numlist)
    
    sort.Slice(copied, func(i, j int) bool {
        a := Distance(copied[i], n)
        b := Distance(copied[j], n)
        if a == b {
            return copied[i] > copied[j]
        } else {
            return a < b
        }
    })
    
    return copied
}